### ISSUE TITLE
Consider removing some node dependencies

### ISSUE THREAD

**Issue by sebastienbarre** [girder/girder#1540]
If possible, that is.
(see [nolanlawson/slow-deps](https://github.com/nolanlawson/slow-deps))

```
girder/girder (webpack)$ slow-deps
Analyzing 58 dependencies...
[====================] 100% 0.0s
---------------------------------------------------------------
| Dependency                        | Time  | Size   | # Deps |
---------------------------------------------------------------
| webpack                           | 37.8s | 14 MB  | 192    |
| jscs                              | 35s   | 26 MB  | 197    |
| webpack-dev-server                | 29.4s | 5.7 MB | 162    |
| css-loader                        | 22.5s | 11 MB  | 115    |
| phantomjs                         | 21.3s | 14 MB  | 98     |
| eslint                            | 20.2s | 9.2 MB | 126    |
| babel-preset-es2015               | 18.4s | 6.6 MB | 93     |
| grunt-contrib-uglify              | 16.8s | 4.3 MB | 85     |
| grunt-npm-install                 | 13.9s | 11 MB  | 2      |
| babel-core                        | 11.7s | 5.9 MB | 53     |
| topojson                          | 11.7s | 6.1 MB | 42     |
| jade                              | 11.5s | 4.4 MB | 40     |
| jsoneditor                        | 10.1s | 11 MB  | 11     |
| grunt-fontello                    | 9.3s  | 9.9 MB | 51     |
| grunt-contrib-jade                | 8.7s  | 3.2 MB | 36     |
| grunt-contrib-stylus              | 6.7s  | 3.2 MB | 33     |
| jshint                            | 6.3s  | 2.8 MB | 23     |
| grunt-contrib-watch               | 5.3s  | 2.5 MB | 18     |
| nib                               | 5s    | 1.6 MB | 21     |
| grunt                             | 4.8s  | 7.6 MB | 30     |
| eonasdan-bootstrap-datetimepicker | 4.5s  | 12 MB  | 3      |
| stylus                            | 4.4s  | 1.4 MB | 20     |
| blanket                           | 4.4s  | 4.3 MB | 5      |
| grunt-cli                         | 3.6s  | 1.3 MB | 11     |
| grunt-shell                       | 3.3s  | 77 KB  | 10     |
| extract-text-webpack-plugin       | 3.3s  | 1.2 MB | 10     |
| babel-loader                      | 2.8s  | 222 KB | 8      |
| grunt-webpack                     | 2.8s  | 1.5 MB | 1      |
| bootstrap-switch                  | 2.8s  | 2.0 MB | 1      |
| as-jqplot                         | 2.8s  | 6.8 MB | 0      |
| event-source                      | 2.8s  | 558 KB | 4      |
| remarkable                        | 2.7s  | 1.5 MB | 5      |
| url-loader                        | 2.6s  | 222 KB | 7      |
| eslint-plugin-underscore          | 2.6s  | 926 KB | 1      |
| style-loader                      | 2.4s  | 168 KB | 6      |
| nopt-grunt-fix                    | 2.4s  | 57 KB  | 3      |
| comment-json                      | 2.3s  | 1.1 MB | 3      |
| grunt-gitinfo                     | 2.2s  | 1.1 MB | 3      |
| stylus-loader                     | 2.2s  | 339 KB | 7      |
| swagger-ui                        | 2.1s  | 4.1 MB | 0      |
| file-loader                       | 2.1s  | 157 KB | 6      |
| jade-loader                       | 1.9s  | 164 KB | 6      |
| backbone                          | 1.8s  | 228 KB | 1      |
| argv                              | 1.7s  | 31 KB  | 0      |
| toposort                          | 1.6s  | 47 KB  | 0      |
| bootstrap                         | 1.6s  | 2.3 MB | 0      |
| jquery                            | 1.6s  | 1.6 MB | 0      |
| eslint-plugin-promise             | 1.3s  | 24 KB  | 0      |
| phantomjs-polyfill                | 1.3s  | 4.0 KB | 0      |
| eslint-config-standard            | 1.2s  | 13 KB  | 0      |
| eslint-plugin-backbone            | 1.2s  | 57 KB  | 0      |
| eslint-config-semistandard        | 1.2s  | 18 KB  | 0      |
| eslint-plugin-standard            | 1.1s  | 75 KB  | 0      |
| underscore                        | 1.1s  | 103 KB | 0      |
| colors                            | 1.1s  | 23 KB  | 0      |
| grunt-file-creator                | 1.1s  | 18 KB  | 0      |
| grunt-contrib-copy                | 1s    | 14 KB  | 0      |
| sprintf-js                        | 1s    | 38 KB  | 0      |
---------------------------------------------------------------
Total time (non-deduped): 6m 30s
Total size (non-deduped): 206 MB
```

**Comment by sebastienbarre** [comment]
- `jscs` (and `jshint`) I find it redundant if using `ESLint` already, but your mileage may vary...,
- I think `grunt-contrib-uglify`, `grunt-contrib-stylus` and `grunt-contrib-jade` I should be able to get rid of in the `webpack` branch,

**Comment by jbeezley** [comment]
We could consider removing jscs and jshint, which only exist as deprecated style checks for plugins.

**Comment by zachmullen** [comment]
Yep, we should remove those on 2.0-integration.

**Comment by sebastienbarre** [comment]
Thanks @jbeezley 
```
Analyzing 54 dependencies...
[====================] 100% 0.0s
---------------------------------------------------------------
| Dependency                        | Time  | Size   | # Deps |
---------------------------------------------------------------
| webpack                           | 43.2s | 14 MB  | 192    |
| webpack-dev-server                | 31.9s | 5.7 MB | 162    |
| css-loader                        | 25.4s | 11 MB  | 115    |
| eslint                            | 20.1s | 9.2 MB | 126    |
| babel-preset-es2015               | 19.6s | 6.6 MB | 93     |
| phantomjs                         | 18.4s | 14 MB  | 98     |
| grunt-contrib-uglify              | 17.9s | 4.3 MB | 85     |
| grunt-npm-install                 | 17.3s | 11 MB  | 2      |
| babel-core                        | 13.8s | 5.9 MB | 53     |
| grunt-fontello                    | 10.4s | 9.9 MB | 51     |
| grunt-contrib-jade                | 10s   | 3.2 MB | 36     |
| topojson                          | 9.2s  | 6.1 MB | 42     |
| jade                              | 7.8s  | 4.4 MB | 40     |
| grunt-contrib-stylus              | 7.6s  | 3.2 MB | 33     |
| jsoneditor                        | 6.7s  | 11 MB  | 11     |
| stylus                            | 5.9s  | 1.4 MB | 20     |
| grunt-contrib-watch               | 5.3s  | 2.5 MB | 18     |
| grunt                             | 5.2s  | 7.6 MB | 30     |
| nib                               | 4.9s  | 1.6 MB | 21     |
| eonasdan-bootstrap-datetimepicker | 4.6s  | 12 MB  | 3      |
| extract-text-webpack-plugin       | 4s    | 1.2 MB | 10     |
| blanket                           | 3.9s  | 4.3 MB | 5      |
| grunt-cli                         | 3.8s  | 1.3 MB | 11     |
| event-source                      | 3.7s  | 558 KB | 4      |
| grunt-shell                       | 3.3s  | 77 KB  | 10     |
| jade-loader                       | 2.9s  | 164 KB | 6      |
| grunt-webpack                     | 2.9s  | 1.5 MB | 1      |
| babel-loader                      | 2.8s  | 222 KB | 8      |
| file-loader                       | 2.8s  | 157 KB | 6      |
| as-jqplot                         | 2.8s  | 6.8 MB | 0      |
| swagger-ui                        | 2.6s  | 4.1 MB | 0      |
| bootstrap-switch                  | 2.5s  | 2.0 MB | 1      |
| eslint-plugin-underscore          | 2.4s  | 926 KB | 1      |
| grunt-gitinfo                     | 2.4s  | 1.1 MB | 3      |
| url-loader                        | 2.2s  | 222 KB | 7      |
| bootstrap                         | 2.2s  | 2.3 MB | 0      |
| remarkable                        | 2.1s  | 1.5 MB | 5      |
| style-loader                      | 2.1s  | 168 KB | 6      |
| stylus-loader                     | 2.1s  | 339 KB | 7      |
| jquery                            | 1.9s  | 1.6 MB | 0      |
| nopt-grunt-fix                    | 1.8s  | 57 KB  | 3      |
| phantomjs-polyfill                | 1.8s  | 4.0 KB | 0      |
| colors                            | 1.7s  | 23 KB  | 0      |
| backbone                          | 1.7s  | 228 KB | 1      |
| eslint-plugin-standard            | 1.6s  | 75 KB  | 0      |
| eslint-plugin-backbone            | 1.6s  | 57 KB  | 0      |
| sprintf-js                        | 1.4s  | 38 KB  | 0      |
| eslint-plugin-promise             | 1.2s  | 24 KB  | 0      |
| eslint-config-semistandard        | 1.2s  | 18 KB  | 0      |
| underscore                        | 1.2s  | 103 KB | 0      |
| eslint-config-standard            | 1.2s  | 13 KB  | 0      |
| grunt-contrib-copy                | 1.1s  | 14 KB  | 0      |
| toposort                          | 1.1s  | 47 KB  | 0      |
| grunt-file-creator                | 993ms | 18 KB  | 0      |
---------------------------------------------------------------
Total time (non-deduped): 6m 0.3s
Total size (non-deduped): 176 MB
```
